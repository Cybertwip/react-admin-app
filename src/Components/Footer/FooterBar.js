import React from 'react';
import {
  Container
 } from 'reactstrap';

export default class FooterBar extends React.Component {
  render() {
    return (
      <Container className="footer">
      <div className="Footer-credits">
        <span className="Footer-credit">Source code and examples released under the <a href="https://github.com/VictorJL/react-web-admin-app/blob/master/LICENSE">MIT</a> license.</span>
      </div>
      </Container>
    );
  }
}
