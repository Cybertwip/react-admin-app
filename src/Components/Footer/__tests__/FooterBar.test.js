import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { mount } from 'enzyme';

import renderer from 'react-test-renderer';

import FooterBar from './../FooterBar';

describe('Main component rendering', () => {
    it('Full-render renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<FooterBar />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('Shallow-render renders without crashing', () => {
      shallow(<FooterBar />);
    });

    it('Render to match snapshot', () => { // 1
      const tree = renderer.create(<FooterBar />).toJSON();
      
      expect(tree).toMatchSnapshot();
    });
});
