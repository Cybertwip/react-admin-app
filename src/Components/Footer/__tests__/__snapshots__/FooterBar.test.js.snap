// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component rendering Render to match snapshot 1`] = `
<div
  className="footer container"
>
  <span
    className="Footer-social"
  >
    <a
      className="twitter-follow-button"
      data-lang="en"
      data-show-count="true"
      href="https://twitter.com/greentwip"
    >
      @greentwip
    </a>
    <a
      className="twitter-share-button"
      data-count="horizontal"
      data-related="Greentwip:Creator of CRUD Web Admin Panel"
      data-text="A showcase of traditionally CRUD problems, easily solved using react."
      data-url="https://github.com/VictorJL/react-web-admin-app"
      data-via="greentwip"
      href="https://twitter.com/share"
    >
      Tweet
    </a>
  </span>
  <div
    className="Footer-credits"
  >
    <span
      className="Footer-credit"
    >
      Created and maintained by 
      <a
        href="https://twitter.com/Greentwip"
      >
        Victor Lopez
      </a>
      . 
    </span>
    <span
      className="Footer-credit"
    >
      Source code and examples released under the 
      <a
        href="https://github.com/VictorJL/react-web-admin-app/blob/master/LICENSE"
      >
        MIT
      </a>
       license.
    </span>
  </div>
</div>
`;
